src/super_gradients/training/processing/processing.py

```diff
@@ -357,6 +371,160 @@ def preprocess_image(self, image: np.ndarray) -> Tuple[np.ndarray, None]:
+    """Rescale image by scaling factor while preserving aspect ratio.
+    The rescaling can be done according to scale_factor, short_size or long_size.
+    If more than one argument is given, the rescaling mode is determined by this order: scale_factor, then short_size,
+    then long_size.
+
+    :param scale_factor: Rescaling is done by multiplying input size by scale_factor:
+        out_size = (scale_factor * w, scale_factor * h)
+    :param short_size:  Rescaling is done by determining the scale factor by the ratio short_size / min(h, w).
+    :param long_size:   Rescaling is done by determining the scale factor by the ratio long_size / max(h, w).
+    """
+
+    def __init__(self, scale_factor: float, short_size: int, long_size: int):
+        self.scale_factor = scale_factor
+        self.short_size = short_size
+        self.long_size = long_size
+
+    def preprocess_image(self, image: np.ndarray) -> Tuple[np.ndarray, SegmentationResizeMetadata]:
+        height, width = image.shape[:2]
+        scale_factor = _compute_scale_factor(self.scale_factor, self.short_size, self.long_size, width, height)
+
+        if scale_factor != 1.0:
+            new_width, new_height = int(scale_factor * width), int(scale_factor * height)
+            resized_image = _rescale_image(image, target_shape=(new_height, new_width))
+
+        return resized_image, SegmentationResizeMetadata(original_shape=(height, width))
+
+    def postprocess_predictions(self, predictions: SegmentationPrediction, metadata: SegmentationResizeMetadata) -> SegmentationPrediction:
+        predictions.segmentation_map = _rescale_image(predictions.segmentation_map.astype(np.uint8), target_shape=metadata.original_shape)
+        predictions.segmentation_map = predictions.segmentation_map.astype(np.int64)
+        return predictions
+
+
+@register_processing(Processings.SegmentationResize)
+class SegmentationResize(Processing):
+    """Resize image to given image dimensions.
+
+    :param output_shape:    output shape will be (output_h, output_w)
+    """
+
+    def __init__(self, output_shape: Tuple[int, int]):
+        self.output_shape = output_shape
+
+    def preprocess_image(self, image: np.ndarray) -> Tuple[np.ndarray, SegmentationResizeMetadata]:
+        height, width = image.shape[:2]
+        image = _rescale_image(image, target_shape=self.output_shape)
+
+        return image, SegmentationResizeMetadata(original_shape=(height, width))
+
+    def postprocess_predictions(self, predictions: SegmentationPrediction, metadata: SegmentationResizeMetadata) -> SegmentationPrediction:
+        predictions.segmentation_map = _rescale_image(predictions.segmentation_map.astype(np.uint8), target_shape=metadata.original_shape)
+        predictions.segmentation_map = predictions.segmentation_map.astype(np.int64)
+        return predictions
+
+
+@register_processing(Processings.SegmentationPadShortToCropSize)
+class SegmentationPadShortToCropSize(Processing):
```

Comment:
Please add unit tests to all newly added classes. The DoD:
- Tests are passing for "normal" inputs.
- Tests are passing for 1x1 images
- Tests are handling a combination of images/params that are edge cases, like `SegmentationPadShortToCropSize(crop_size=(512,512)), input is 640,640`. Ideally behavior of transforms in such tests should be documented well.

Reply:
done (2dc024f345594194a221854c69e2525307c9a516)

please note that this transform doesn't crop the image to into "crop_size", it only "enlarges" the image if needed (by padding)
if "crop_size" is larger than image size - the image will be padded into "crop size"
if "crop size" is smaller than image size - no change will be made  
the equivalent processing behaves in the same way

@Louis-Dupont @shaydeci - FYI

Reply:
Louis opened a ticket regarding transform/processing name (called crop but doesn't crop)